rpc/lib/server/handlers.go

```diff
@@ -434,8 +433,8 @@ type wsConnection struct {
 	// Send pings to server with this period. Must be less than readWait, but greater than zero.
 	pingPeriod time.Duration
 
-	// object that is used to subscribe / unsubscribe from events
-	eventSub types.EventSubscriber
+	// callback which is called upon disconnect
+	onDisconnect func(remoteAddr string)
```

Comment:
What prompted this?

Reply:
Bad design? The fact that low-level rpc library is highly-coupled with the `types` package and knows there's a pubsub (thing it can use to Subscribe/Unsubscribe) is a bad design. I realized we can Unsubscribe via a disconnect callback and Subscribe in the calling code (`lite/proxy`).

Reply:
Oh goodness, excellent point!